skbio/alignment/_util.py

```diff
@@ -0,0 +1,70 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import numpy as np
+
+
+def pairwise_align_score(path, seqs, match=2, mismatch=-3, gap_open=-5, gap_extend=-2):
+    """Calculate alignment score of a pairwise alignment."""
+    # The current code only accepts an alignment path and the original sequences. It is
+    # optimized for this data structure.
+    # TODO: It should also accept a tabular MSA object. For optimal performance, it
+    # should be a separate function. A dispatcher will unite the two functions.
+    if path.shape[0] != 2:
+        raise ValueError("Only pairwise alignment is accepted.")
+
+    # step 1: calculate gap penalties
+    # based on the number and lengths of gap segments
+    # agnostic of the sequences
+    gap_segs = path.states != 0
+    n_gap_opens = gap_segs.sum()
+    n_gap_extends = (gap_segs * path.lengths).sum() - n_gap_opens
+
+    # step 2: calculate match & mismatch scores
+    # based on the characters within the non-gap segments
+    # This will work for a strictly pairwise alignment, where it is certain that
+    # gaps will not occur in the same position in both sequences.
+    seq1, seq2 = (x._bytes for x in seqs)
```

Comment:
@qiyunzhu This code is much simpler than the original, which has been commented out below. I think the original intended to account for situations where there are gaps in the same position in the two sequences. My understanding is that in a purely pairwise alignment, this will never occur. Is that correct? I feel this function should just be about a straightforward pairwise alignment score. Handling other cases could be put into the other functions in this file. What do you think?